packages/react-core/src/components/Card/Card.tsx

```diff
@@ -8,7 +8,7 @@ export interface CardProps extends React.HTMLProps<HTMLElement> {
   /** Additional classes added to the Card */
   className?: string;
   /** Sets the base component to render. defaults to article */
-  component?: keyof JSX.IntrinsicElements;
```

Comment:
this is rendered as DOM element like 'div', should not be react node

Reply:
In our implementation, we do `<Component>` which compiles to `React.createElement(Component)` where `Component` is of type `React.ReactNode`. We use `React.ReactNode` which is inclusive of `keyof JSX.IntrinsicElements` so it's not a breaking change.

Reply:
that means you could pass `component={<Alert title="test" />}` which would not fare well here

Reply:
Sure, this is outside the scope of this PR anyways.